src/index.rs

```diff
@@ -340,38 +340,47 @@ impl ResourceIndex {
 
     pub fn update_one(
         &mut self,
-        path_buf: CanonicalPathBuf,
+        path: &dyn AsRef<Path>,
         old_id: ResourceId,
     ) -> Result<IndexUpdate> {
         log::debug!("Updating a single entry in the index");
-
-        let path = path_buf.as_canonical_path();
+        
+        // the caller must have ensured that the path is exist
+        if !path.as_ref().exists() {
+            return Err(ArklibError::Path(
+                "Path does not exist".into(),
+            ));
+        }
```

Comment:
We actually need to support this case as normal scenario. When we delete a resource, we need to update the index. Consumer app first deletes the resource then calls `update_one` with its path, which already doesn't exist.

Reply:
noted.